features/net/network-socket/stack_mem.h

```diff
@@ -0,0 +1,84 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_STACK_MEM_H
+#define MBED_STACK_MEM_H
+
+/**
+ * Stack memory module
+ *
+ * This interface provides abstraction for memory modules used in different IP stacks (often to accommodate zero copy).
+ * Emac interface may be required to accept output packets and provide received data using this stack specific API.
+ * This header should be implemented for each IP stack, so that we keep emacs module independent.
+ */
+typedef void stack_mem_t;
+typedef void stack_mem_chain_t;
+
+/**
+ * Allocates stack memory
+ *
+ * @param  size  Size of memory to allocate
+ * @param  align Memory alignment requirements
+ * @return       Allocated memory struct, or NULL in case of error
+ */
+stack_mem_t *stack_mem_alloc(uint32_t size, uint32_t align);
```

Comment:
There should also be an allocator context parameter in this API, so the memory allocator does not require global state.